src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -2,18 +2,47 @@
 
 import core.basesyntax.Storage;
 
+@SuppressWarnings("unchecked") // do not remove this line
+
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ARRAY_ELEMENTS = 10;
+    private K key;
+    private V value;
+    private K[] keyArray = (K[]) new Object[MAX_ARRAY_ELEMENTS];
+    private V[] valueArray = (V[]) new Object[MAX_ARRAY_ELEMENTS];
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < keyArray.length; i++) {
+            if ((keyArray[i] == null && valueArray[i] == null)
+                    || (key != null && key.equals(keyArray[i]))
+                    || (key == null && keyArray[i] == null)) {
+                keyArray[i] = key;
+                valueArray[i] = value;
+                return;
+            }
+        }
+        throw new RuntimeException("Can't put the data into a Storage - Storage is full ");
```

Comment:
```suggestion
        keyArra[size] = key;
        valueArray[size] = value;
        size++;
```
if the key is not present, you put the key and value on the last free position
if you want to throw an exception for the situation when the storage is full, you can add a check at the beginning of this method. You should check if size == MAX_ARRAY_ELEMENTS